File: dojo/mik.s/FizzBuzz/CSharp/05-02-2012 WhiteBelt/FizzBuzz/Printer.cs
using System;
using System.Collections.Generic;

namespace FizzBuzz {
    public class Printer {
        private ILinePrinter _linePrinter;

        public Printer(ILinePrinter linePrinter) {
            if (linePrinter == null) {
                throw new ArgumentNullException("linePrinter");
            }

            _linePrinter = linePrinter;
        }

        public IEnumerable<string> PrintLines(uint numberOfLines) {
            if (numberOfLines == 0) {
                throw new ArgumentException();
            }

            for (uint line=1; line<=numberOfLines; line++) {
                yield return _linePrinter.PrintLine(line);
            }
        }
    }
}

File: dojo/mik.s/FizzBuzz/Javascript/05-02-2012 WhiteBelt/src/Printer.js
function Printer(linePrinter) {
	if (!linePrinter) {
		throw new Error("linePrinter may not be null");
	}
	
	if (!linePrinter.printLine) {
		throw new Error("linePrinter must have a function called 'printLine'");
	}
	
	this.linePrinter = linePrinter;
}

Printer.prototype.printLines = function(numberOfLines) {
	if (isNaN(numberOfLines) || numberOfLines <= 0) {
		throw new Error("numberOfLines must be a number greater than zero");
	}
	
	var lines = new Array();
	
	for (var i=1; i<=numberOfLines; i++) {
		var line = this.linePrinter.printLine(i);
		lines[i-1] = line;
	}
	
	return lines;
}